### PR TITLE
HWKAPM-574 Fix NPE in communication summary graph builder due to mism…

### PR DESCRIPTION
**Pull request by objectiser**
…atched source/target URLs

https://issues.jboss.org/browse/HWKAPM-574

### CODE REVIEW

**Comment by objectiser** [comment]
Reviewer note:
* issue was caused due to mismatch in the expected source/target URIs used when building the graph - as they were using a different notion of 'operation' - so consolidated on using a single 'deriveOperation' helper
* other problem was that one of the reported spans was not being recognized as a http span, as it was trying to search for http response code which was not present, so changed the check to be based on presence of http.url binary annotation.

@pavolloffay could you review when you have time?

**Comment by objectiser** [comment]
btw - jira hwkapm 571 should hopefully replace much of this hardcoded analysis of the binary annotations to make it more flexible to deal with new (or user specific) annotations.

File: server/api/src/main/java/org/hawkular/apm/server/api/model/zipkin/Span.java [hunk 1]
```diff
@@ -222,14 +222,6 @@ public URL url() {
         return null;
     }
 
-    public String operation() {
```
Comment (pavolloffay) [inline server/api/src/main/java/org/hawkular/apm/server/api/model/zipkin/Span.java]: :+1: I was looking at it yesterday.. :+1:

**Comment by pavolloffay** [comment]
LGTM :+1:

**Pull request merged**